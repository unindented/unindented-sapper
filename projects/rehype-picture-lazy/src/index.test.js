const raw = require("rehype-raw");
const stringify = require("rehype-stringify");
const remark2rehype = require("remark-rehype");
const markdown = require("remark-parse");
const unified = require("unified");
const picture = require(".");

const process = (text, options) =>
  new Promise((resolve, reject) => {
    unified()
      .use(markdown)
      .use(remark2rehype, { allowDangerousHTML: true })
      .use(raw)
      .use(picture, options)
      .use(stringify)
      .process(text, (err, contents) => {
        if (!err) {
          resolve(contents.contents);
        } else {
          reject(err);
        }
      });
  });

describe("remark-picture-lazy", () => {
  it("leaves image alone if not configured", () => {
    const text = `
Here's an image:

<img src="foo.png" alt="Some image" width="640" height="480" />
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });

  it("wraps image in picture if configured", () => {
    const text = `
Here's another image:

<img src="foo.png" alt="Some image" width="640" height="480" />
`;
    return expect(
      process(text, { png: { webp: "image/webp" } })
    ).resolves.toMatchSnapshot();
  });
});
