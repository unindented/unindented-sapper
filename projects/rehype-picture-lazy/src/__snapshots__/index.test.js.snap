// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark-picture-lazy leaves image alone if not configured 1`] = `
"<p>Here's an image:</p>
<img src=\\"foo.png\\" alt=\\"Some image\\" width=\\"640\\" height=\\"480\\">"
`;

exports[`remark-picture-lazy wraps image in picture if configured 1`] = `
"<p>Here's another image:</p>
<picture><source srcset=\\"foo.webp\\" type=\\"image/webp\\" data-width=\\"640\\" data-height=\\"480\\" data-placeholder=\\"data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgNjQwIDQ4MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj48L3N2Zz4=\\"><img src=\\"foo.png\\" alt=\\"Some image\\" width=\\"640\\" height=\\"480\\"></picture>"
`;
